File: TP1/main.py
from src.etudiant import Etudiant
from src.enseignant import Enseignant

def main():
    e = Etudiant("Alice", 20, "ETU001", 15.5)
    ens = Enseignant("Dr. Martin", 45, "Informatique", 3500.0)
    
    personnes = [e, ens]
    
    for p in personnes:
        print(p.afficher_details())

if __name__ == "__main__":
    main()

File: TP1/java/main.java
import java.util.ArrayList;
import java.util.List;

public class Main {
    public static void main(String[] args) {
        List<Personne> personnes = new ArrayList<>();
        personnes.add(new Etudiant("Alice", 20, "ETU001", 15.5));
        personnes.add(new Enseignant("Dr. Martin", 45, "Informatique", 3500.0));

        for (Personne p : personnes) {
            System.out.println(p.afficherDetails());
        }
    }
}
